### DISCUSSION TITLE
Welcome to -MPMDv2-modifications-and-fixes Discussions!

### DISCUSSION THREAD

**Discussion by Mclute0** [Mclute0/-MPMDv2-modifications-and-fixes#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by theTechnoWeenie** [comment]
Hello! I'm Sam and I just wanted to say thank you for putting this up! 

I'd been looking all over for someone that was having problems with the USB on the MPMDv2 and I was resigned to just living with it. The printer was working fine, so I didn't wanna take apart the whole board, but you having traced the usb connection and found the nRst pin silly decision was EXACTLY what I was looking for. I'm up and running with pronterface now and can finally tune the printer the way it deserves. Thanks!

**Reply by Mclute0** [reply]
glad to help out. It is a shame that Monoprice is so clueless with customer support.

**Comment by adamsyed10** [comment]
I'm unable to get pronterface to connect to the mini delta v2. I would like to factory reset it via gcode, "m502", but am not sure how to.  Wiibuilder doesn't let me send gcode commands. Thanks so much for your informative writeup, please let me know if you are able to find a way.

**Reply by piberry** [reply]
Create a text file and put whatever gcode you want in it, then just change the extension to .gcode, copy to sd card and simply print it.

**Comment by adamsyed10** [comment]
I have an extra one and am willing to ship it to you if you are able to find a way to factory reset mine.

**Reply by piberry** [reply]
You can´t really fully factory reset. At least, if you were connected to wifi, that information stays forever.
The machine runs on marlin firmware but the ESP32LCD has a separate firmware (closed source) which stores wifi credentials and print time for example. This is why sending M502 doesn´t help here because it will factory reset the marlin part only. M502 does nothing on the ESP32 chip that controls the LCD.

If you really want to "factory reset" the ESP32LCD, check this forked repo:
https://github.com/piberry/-MPMDv2-modifications-and-fixes

You need soldering skills and a FTDI for this, the rest is easy.

This is the partiton that holds the data and wifi information on the ESP32 chip:
entry 0:
label : nvs
offset : 0x9000
length : 16384
type : 1 [DATA]
sub type : 2 [WIFI]

It´s also possible to flash just the nvs partiton with a blanked file to erase the data.

If you want to do it and need assistance, hit me up. 
The facebook group is a better place to ask your questions:
https://www.facebook.com/groups/581769846127833
